### PR TITLE
T dh 600 case work orders

### PR DESCRIPTION
**Pull request by byarsawich**
Work order component for the case intake page.

### CODE REVIEW

File: src/aura/AddFee/AddFee.cmp [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    <aura:attribute name="addedFee" type="base_toc__Fee__c" default="{ 'sobjectType' : 'base_toc__Fee__c' }"/>
+
+    <aura:handler name="change" value="{!v.addedFee}" action="{!c.handleChange}" />
+
+    <aura:registerEvent name="removeAddOn" type="c.RemoveAddOn" />
+    <aura:registerEvent name="createAddOn" type="c.CreateAddOn" />
+
+    <lightning:layout horizontalAlign="center" verticalAlign="stretch" multipleRows="true">
+        <lightning:layoutItem size="7" flexibility="auto" padding="around-small" title="Product" >
+            <!-- TODO: need to add an is valid formula to the base package to chekc the dates because < and > can't be included in the query -->
```
Comment (Fitchburgh) [inline src/aura/AddFee/AddFee.cmp]: What "and" are you trying to use?
if you set the queryCondition to an attribute and set it in the controller or helper and should work. 
Ex: component.set("v.wt", "WorkType.RecordTypeId=\'" + response.getReturnValue() + "\' AND WorkType.Inactive__c = false");

File: src/aura/AddProduct/AddProductHelper.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+({
+	getPicklist : function(component) {
```
Comment (Fitchburgh) [inline src/aura/AddProduct/AddProductHelper.js]: Does this need to do anything?

File: src/aura/CaseIntakeWork/CaseIntakeWork.cmp [hunk 3]
```diff
@@ -0,0 +1,99 @@
+<aura:component controller="CaseIntakeWorkController" access="global" implements="flexipage:availableForAllPageTypes">
```
Comment (Fitchburgh) [inline src/aura/CaseIntakeWork/CaseIntakeWork.cmp]: should we allow the url in here?  I'm sure someone will want to "launch case intake from an address/asset/work type" at some point in the future - just proof it in now?

Comment (Fitchburgh) [reply src/aura/CaseIntakeWork/CaseIntakeWork.cmp]: also there's some weird spacing in this cmp, just check it over

File: src/aura/CaseIntakeWork/CaseIntakeWorkHelper.js [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+		});
+		action.setCallback(this, function(response){
+			console.log('in the callback');
+			var acsHelper = component.find('acsHelper');
+			var state = response.getState();
+			if (state === 'SUCCESS') {
+				var result = JSON.parse(response.getReturnValue());
+				console.log(result);
+				if(!result) {
+					acsHelper.showToast('error', 'There Was A Problem in Work Order Creation', 'Please stop screwing up your input.  Thank you for your pantronage.');
```
Comment (Fitchburgh) [inline src/aura/CaseIntakeWork/CaseIntakeWorkHelper.js]: lol

File: src/classes/CaseIntakeWorkController.cls [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+			Database.rollback(sp);
+			deleteCase(newWO.CaseId);
+			return null;
+		}
+
+		return JSON.serialize(newWO);
+	}
+
+	private static void deleteCase(Id caseId) {
+		List<Case> c =
```
Comment (Fitchburgh) [inline src/classes/CaseIntakeWorkController.cls]: list necessary? it should be a unique id, right?

File: src/classes/Test_CaseIntakeWorkController.cls [hunk 6]
```diff
@@ -0,0 +1,7 @@
+/*
+ * Created by byarsawi on 10/5/2018.
+ */
+@IsTest
+private class Test_CaseIntakeWorkController {
+
```
Comment (Fitchburgh) [inline src/classes/Test_CaseIntakeWorkController.cls]: todo 👍  ha - we could have Cailen take a stab at this - he's been asking about test classes

File: src/aura/AddProduct/AddProductHelper.js (hunk 2 above)
Comment (byarsawich) [reply src/aura/AddProduct/AddProductHelper.js]: yes it does

File: src/aura/CaseIntakeWork/CaseIntakeWork.cmp (hunk 3 above)
Comment (byarsawich) [reply src/aura/CaseIntakeWork/CaseIntakeWork.cmp]: not on this one, I'd prefer to send anything for a stand alone request to work order lightning

File: src/aura/AddFee/AddFee.cmp (hunk 1 above)
Comment (byarsawich) [reply src/aura/AddFee/AddFee.cmp]: Finally remembered what this was.  it's not that I could get AND in the query, it was I couldn't use greater than > or less than < in the query.

Comment (Fitchburgh) [reply src/aura/AddFee/AddFee.cmp]: Right on.

File: src/aura/CaseIntakeWork/CaseIntakeWork.cmp (hunk 3 above)
Comment (Fitchburgh) [reply src/aura/CaseIntakeWork/CaseIntakeWork.cmp]: 10-4 makes sense

**Pull request merged**